### PR TITLE
feat/home screen

### PR DESCRIPTION
**Pull request by sieuhuflit**
## Overview

- [x] Home screen 
- [x] Display post card
- [x] Handle display default avatar if no avatar
- [x] Handle display default event if no event image

## Demo

Simulator Screenshot - iPhone 14 - 2023-08-03 at 22 00 01

### CODE REVIEW

File: lib/core/data/post/newsfeed_repository_impl.dart [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import 'package:injectable/injectable.dart';
+
+@LazySingleton(as: NewsfeedRepository)
+class NewsfeedRepositoryImpl implements NewsfeedRepository {
+  final _client = getIt<AppGQL>().client;
+
+  @override
+  Future<Either<Failure, Newsfeed>> getNewsfeed(
+      {GetNewsfeedInput? input}) async {
+    if (input?.offset == 0) {
```
Comment (mthinh) [inline lib/core/data/post/newsfeed_repository_impl.dart:20]: Wonder why we need to separate into 2 cases ?

File: lib/core/domain/post/input/get_posts_input.dart [hunk 2]
```diff
@@ -13,6 +13,7 @@ class GetPostsInput with _$GetPostsInput {
     @JsonKey(name: 'created_at', includeIfNull: false) GetPostCreatedAtInput? createdAt,
     @JsonKey(includeIfNull: false) int? skip,
     @JsonKey(includeIfNull: false) int? limit,
+    @JsonKey(includeIfNull: false) int? offset,
```
Comment (mthinh) [inline lib/core/domain/post/input/get_posts_input.dart:16]: does get post support offset?

File: lib/core/presentation/pages/home/views/list/home_newsfeed_list.dart [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+              child: EmptyList(emptyText: t.notification.emptyNotifications),
+            );
+          }
+          return ListView.separated(
+            itemBuilder: (ctx, index) => index == newsfeed.length
+                ? const SizedBox(height: 80)
+                : Padding(
+                    padding: EdgeInsets.symmetric(horizontal: Spacing.small),
+                    child: PostProfileCard(
+                        key: Key(newsfeed[index].id), post: newsfeed[index]),
```
Comment (mthinh) [inline lib/core/presentation/pages/home/views/list/home_newsfeed_list.dart:35]: for multiple properties widget, please add a comma before `)` , which will the formatter in to break into lines. which easier to watch

File: lib/core/presentation/widgets/event/event_post_card_widget.dart [hunk 4]
```diff
@@ -137,9 +156,13 @@ class EventPostCard extends StatelessWidget {
...
           TextSpan(
             text: "${event.title}\n",
+            style: Typo.medium.copyWith(fontFamily: FontFamily.circularStd),
             children: [
               TextSpan(
-                style: Typo.small.copyWith(color: colorScheme.onSurface, height: 1.5),
+                style: Typo.small.copyWith(
+                    color: colorScheme.onSurface,
+                    height: 1.5,
+                    fontFamily: FontFamily.circularStd),
```
Comment (mthinh) [inline lib/core/presentation/widgets/event/event_post_card_widget.dart:165]: please help add comma at the end before the )

File: lib/core/presentation/widgets/event/event_post_card_widget.dart [hunk 5]
```diff
@@ -152,6 +175,12 @@ class EventPostCard extends StatelessWidget {
   }
 
   Container _buildEventPhoto() {
+    final imageUrl = eventPhoto != null
+        ? ImageUtils.generateUrl(
+            file: eventPhoto,
+            imageConfig: ImageConfig.eventPhoto,
+          )
+        : '${AppConfig.assetPrefix}/assets/images/no_photo_event.png';
```
Comment (mthinh) [inline lib/core/presentation/widgets/event/event_post_card_widget.dart:183]: when ever you add new image or icon, use ./build_runner.sh, fluttergen will generate assets for you, 
the usage will be like 
`Assets.images.noPhotoEvent.image()`
You can check the usage with other svg

File: lib/core/presentation/widgets/home/what_on_your_mind_input.dart [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+      },
+      child: Container(
+        padding: EdgeInsets.symmetric(horizontal: 15, vertical: 15),
+        decoration: BoxDecoration(
+            borderRadius: BorderRadius.circular(12), color: themeColor.outline),
+        child: Row(
+          children: [
+            Expanded(
+              child: Text(
+                "What's on your mind, Joseph?",
```
Comment (mthinh) [inline lib/core/presentation/widgets/home/what_on_your_mind_input.dart:24]: please use i18n, with real user name

File: lib/core/presentation/widgets/post/post_profile_card_widget.dart [hunk 7]
```diff
@@ -58,24 +63,31 @@ class PostProfileCard extends StatelessWidget {
                 Row(
                   mainAxisSize: MainAxisSize.max,
                   children: [
-                    Text(postUsername),
+                    Text(postName,
+                        style: Typo.medium
+                            .copyWith(fontFamily: FontFamily.circularStd)),
```
Comment (mthinh) [inline lib/core/presentation/widgets/post/post_profile_card_widget.dart:68]: The design for home screen not update the font family yet, please don't use this font. 
Base font already setup in theme

Comment (mthinh) [reply lib/core/presentation/widgets/post/post_profile_card_widget.dart:68]: btw, please add the comma

File: lib/core/presentation/widgets/post/post_profile_card_widget.dart [hunk 8]
```diff
@@ -58,24 +63,31 @@ class PostProfileCard extends StatelessWidget {
...
+                        '  •  ${timeago.format(postCreatedAt!)}',
+                        style:
+                            Typo.medium.copyWith(color: colorScheme.onSurface),
                       ),
                     Spacer(),
                     ThemeSvgIcon(
                       color: colorScheme.onSurface,
-                      builder: (filter) => Assets.icons.icMoreHoriz.svg(colorFilter: filter),
+                      builder: (filter) => Assets.icons.icMoreHoriz
+                          .svg(colorFilter: filter, width: 18, height: 18),
```
Comment (mthinh) [inline lib/core/presentation/widgets/post/post_profile_card_widget.dart:79]: please add the comma

File: lib/core/presentation/widgets/post/post_profile_card_widget.dart [hunk 9]
```diff
@@ -58,24 +63,31 @@ class PostProfileCard extends StatelessWidget {
...
                 ),
                 if (postText.isNotEmpty) ...[
                   SizedBox(height: Spacing.superExtraSmall),
                   Text(
                     postText,
-                    style: Typo.medium.copyWith(color: colorScheme.onSurface),
+                    style: Typo.medium.copyWith(
+                        color: colorScheme.onSurface,
+                        fontWeight: FontWeight.w400,
+                        fontFamily: FontFamily.circularStd),
```
Comment (mthinh) [inline lib/core/presentation/widgets/post/post_profile_card_widget.dart:90]: please add comma and remove the font family

File: lib/core/presentation/widgets/post/post_profile_card_widget.dart [hunk 10]
```diff
@@ -116,4 +129,60 @@ class PostProfileCard extends StatelessWidget {
...
+                  ),
+                ),
+                SizedBox(width: 3),
+                Text(reactions != null ? '$reactions' : '',
+                    style: Typo.small.copyWith(
+                        fontFamily: FontFamily.circularStd,
+                        color: colorScheme.onSecondary)),
+              ],
+            ),
+            SizedBox(width: 12),
```
Comment (mthinh) [inline lib/core/presentation/widgets/post/post_profile_card_widget.dart:158]: use Spacing.xSmall

File: lib/core/presentation/widgets/home/what_on_your_mind_input.dart [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+            GestureDetector(
+              onTap: () {
+                // Add your edit icon onTap logic here
+                // For example, show a dialog or navigate to the edit screen.
+                print("Edit icon tapped!");
+              },
+              child: ThemeSvgIcon(
+                color: themeColor.onSecondary,
+                builder: (filter) => Assets.icons.icEdit
+                    .svg(colorFilter: filter, width: 18, height: 18),
```
Comment (mthinh) [inline lib/core/presentation/widgets/home/what_on_your_mind_input.dart:37]: Please add comma

File: lib/core/presentation/widgets/post/post_profile_card_widget.dart [hunk 12]
```diff
@@ -116,4 +129,60 @@ class PostProfileCard extends StatelessWidget {
...
+                    colorFilter: filter,
+                    width: 18,
+                    height: 18,
+                  ),
+                ),
+                SizedBox(width: 3),
+                Text(comments != null ? '$comments' : '',
+                    style: Typo.small.copyWith(
+                        fontFamily: FontFamily.circularStd,
+                        color: colorScheme.onSecondary)),
```
Comment (mthinh) [inline lib/core/presentation/widgets/post/post_profile_card_widget.dart:173]: please add comma and remove font family

File: lib/core/utils/avatar_utils.dart [hunk 13]
```diff
@@ -0,0 +1,39 @@
+import 'package:app/core/config.dart';
+import 'package:app/core/domain/common/entities/common.dart';
+import 'package:app/core/domain/user/entities/user.dart';
+import 'package:app/core/utils/image_utils.dart';
+
+class AvatarUtils {
+  static String randomImage(
+      String id, int length, String Function(int) template) {
```
Comment (mthinh) [inline lib/core/utils/avatar_utils.dart:8]: please add comma for line break if multiple args

File: lib/core/utils/avatar_utils.dart [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+  static String randomUserImage(String id) {
+    return randomImage(
+      id,
+      10,
+      (num) =>
+          '${AppConfig.assetPrefix}/assets/images/avatars/lemonade_davatar_${num}.png',
+    );
+  }
+
+  static String getAvatarUrl({User? user, bool useRandomUserImage = true}) {
```
Comment (mthinh) [inline lib/core/utils/avatar_utils.dart:26]: add comma to have line-break for multiple args

**Review by mthinh — CHANGES_REQUESTED** [review]
Please help recheck <3

File: lib/core/domain/post/input/get_posts_input.dart (hunk 2 above)
Comment (sieuhuflit) [reply lib/core/domain/post/input/get_posts_input.dart:16]: Mistake

File: lib/core/presentation/widgets/event/event_post_card_widget.dart (hunk 5 above)
Comment (sieuhuflit) [reply lib/core/presentation/widgets/event/event_post_card_widget.dart:183]: Yep, but this one is using the dynamic `no_photo_event.png` load from our assets server, so don't need store that png in app (increase size)

**Comment by sieuhuflit** [comment]
Resolved, pls help check @mthinh

File: lib/core/presentation/widgets/home/what_on_your_mind_input.dart [hunk 15]
```diff
@@ -21,14 +21,12 @@ class WhatOnYourMindInput extends StatelessWidget {
           children: [
             Expanded(
               child: Text(
-                "What's on your mind, Joseph?",
+                t.home.what_on_your_mind,
```
Comment (mthinh) [inline lib/core/presentation/widgets/home/what_on_your_mind_input.dart:24]: currently we use camelCase 😅

File: lib/core/presentation/widgets/post/post_profile_card_widget.dart [hunk 16]
```diff
@@ -63,9 +63,10 @@ class PostProfileCard extends StatelessWidget {
                 Row(
                   mainAxisSize: MainAxisSize.max,
                   children: [
-                    Text(postName,
-                        style: Typo.medium
-                            .copyWith(fontFamily: FontFamily.circularStd)),
+                    Text(
+                      postName,
+                      style: Typo.medium.copyWith(),
```
Comment (mthinh) [inline lib/core/presentation/widgets/post/post_profile_card_widget.dart:68]: no copyWith needed

**Review by mthinh — CHANGES_REQUESTED** [review]
Please help recheck again

**Comment by mthinh** [comment]
LGTM 👍

**Review by mthinh — APPROVED** [review]

**Pull request merged**